src/EntityGraphQL/Schema/SchemaGenerator.cs

```diff
@@ -1,187 +1,206 @@
-        }
-
-        private static object GetGqlArgs(ISchemaProvider schema, IField field, string noArgs = "")
-        {
-            if (field.Arguments == null || !field.Arguments.Any() || field.ArgumentsAreInternal)
-                return noArgs;
-
-            var all = field.Arguments.Select(f => schema.SchemaFieldNamer(f.Key) + ": " + f.Value.Type.GqlTypeForReturnOrArgument);
-
-            var args = string.Join(", ", all);
-            return string.IsNullOrEmpty(args) ? string.Empty : $"({args})";
-        }
-
-        private static object GetDirectiveArgs(ISchemaProvider schema, IDirectiveProcessor directive)
-        {
-            var args = directive.GetArguments(schema);
-            if (args == null || !args.Any())
-                return string.Empty;
-
-            var allArgs = string.Join(", ", args.Select(f => f.Name + ": " + f.Type.GqlTypeForReturnOrArgument));
-            return string.IsNullOrEmpty(allArgs) ? string.Empty : $"({allArgs})";
-        }
-
-        private static string OutputSchemaType(ISchemaProvider schema, ISchemaType schemaType)
-        {
-            var sb = new StringBuilder();
-
-            if (!string.IsNullOrEmpty(schemaType.Description))
-                sb.AppendLine($"\"\"\"{EscapeString(schemaType.Description)}\"\"\"");
-
-
-            if (schemaType.GqlType == GqlTypeEnum.Union)
-            {
-                if (schemaType.PossibleTypes.Count == 0)
-                {
-                    return string.Empty;
-                }
-
-                sb.AppendLine($"union {schemaType.Name} = {string.Join(" | ", schemaType.PossibleTypes.Select(i => i.Name))}");
-                return sb.ToString();
-            }
-
-            var type = schemaType.GqlType switch
-            {
-                GqlTypeEnum.Input => "input",
-                GqlTypeEnum.Interface => "interface",
-                GqlTypeEnum.Union => "union",
-                _ => "type"
-            };
-
-            var implements = "";
-            if (schemaType.BaseTypes != null && schemaType.BaseTypes.Count > 0)
-            {
-                implements += $" implements {string.Join(" & ", schemaType.BaseTypes.Select(i => i.Name))}";
-            }
-
-            sb.AppendLine($"{type} {schemaType.Name}{implements}{GetDirectives(schemaType.Directives)} {{");
-
-            foreach (var field in schemaType.GetFields().OrderBy(s => s.Name))
-            {
-                if (field.Name.StartsWith("__"))
-                    continue;
-                if (!string.IsNullOrEmpty(field.Description))
-                    sb.AppendLine($"\t\"\"\"{EscapeString(field.Description)}\"\"\"");
-                sb.AppendLine($"\t{schema.SchemaFieldNamer(field.Name)}{GetGqlArgs(schema, field)}: {field.ReturnType.GqlTypeForReturnOrArgument}{GetDirectives(field.Directives)}");
-            }
-            sb.AppendLine("}");
-
-            return sb.ToString();
-        }
-    }
+using System;

```

Comment:
Why does it show this file 100% changed? Makes it hard to see what did change!

Any highlights?

Reply:
curious, i must have accidentally formatted it differently. will see if I can clean it up.

there should only be about 2 lines different where it serializes the default value, that said i need to write a custom serializer i think.

Reply:
![image](https://user-images.githubusercontent.com/533131/194029368-6b0d7d31-9599-43fe-96eb-162938b3f0de.png)

and GetArgDefaultValue method is entirely new